utils/debug/PointCloudDebug.js

```diff
@@ -10,39 +10,36 @@ function isInHierarchy(elt, hierarchyNode) {
 
 export default {
     initTools(view, layer, datUi) {
+        const update = () => view.notifyChange(layer);
         layer.debugUI = datUi.addFolder(`${layer.id}`);
 
-        layer.debugUI.add(layer, 'sseThreshold').name('SSE threshold')
-            .onChange(() => view.notifyChange(layer));
-        layer.debugUI.add(layer, 'octreeDepthLimit', -1, 20).name('Depth limit')
-            .onChange(() => view.notifyChange(layer));
-        layer.debugUI.add(layer, 'pointBudget', 1, 15000000).name('Max point count')
-            .onChange(() => view.notifyChange(layer));
+        layer.debugUI.add(layer.material, 'visible').name('Visible').onChange(update);
```

Comment:
I missed this before the merge, but this breaks other parts of itowns which expect `layer.visible`:
at the very least the test here https://github.com/iTowns/itowns/blob/master/src/Core/MainLoop.js#L111 is broken :(

It also has a different behavior than `3dtiles` layer which already supported having a layer.material property: in the 3dtiles case the material is cloned and layer.material is never copied in object.material.

Btw, https://github.com/iTowns/itowns/blob/master/src/Core/MainLoop.js#L111 is a bad idea as well IMHO: a user can't set one parameter for a given point object material because it will always be overriden with the one from the layer (so the overlay color feature can now only be enabled globally).

Reply:
For the rest of the comment, I left `points.onBeforeRender` untouched just for that use case :).

That way, I think you can still get your per-tile functionality, while fixing the bug present before the merge : if you change styling properties on the layer, then it will only apply to newly created tiles, and not to previously created tiles (just take an earlier commit, change the point size and move around to see that in action).

That being said, maybe we should open an issue to agree and document properly a common strategy for the style/material of all layers.

Reply:
I agree there was a bug before the merge.

I disagree that `onBeforeRender` is an acceptable substitute tough: having to register a callback for each object you want to customize vs modifying their material once is quite different (looks very much like pushing the complexity on the user of itowns to ease the implementation in itowns core).

That being said, this is a minor feature, so we can discuss this in an issue and try to find a solution that works for all kind of layer.

In the meantime please fix the other issues, especially the `visible`. Thanks!